pkg/kubelet/kubelet_pods.go

```diff
@@ -435,6 +435,21 @@ func (kl *Kubelet) GetPodCgroupParent(pod *v1.Pod) string {
 	return cgroupParent
 }
 
+// GetHostNameForKernel gets hostname value to set in the hostname field (the nodename field of struct utsname) of the pod.
+func (kl *Kubelet) GetHostNameForKernel(hostname string, hostDomainName string, hostnameFQDN bool) (string, error) {
+	kernelhostname := hostname
+	if len(hostDomainName) > 0 && hostnameFQDN {
+		fqdnHostname := fmt.Sprintf("%s.%s", hostname, hostDomainName)
+		// FQDN has to be 63 chars to fit in hostname (specification is 64bytes which is 63 chars and the null terminating char).
+		const hostnameMaxLen = 63
+		if len(fqdnHostname) > hostnameMaxLen {
+			return "", fmt.Errorf("Failed to set FQDN in hostname, Pod hostname %s is too long (%d characters is the limit)", fqdnHostname, hostnameMaxLen)
```

Comment:
Failed to get FQDN from pod  hostname

Reply:
thanks, rephrasing

Reply:
Also say how many characters we have, eg "max is 64, requested is 72"

Reply:
ok, adding requested length too